### PR TITLE
Add onboarding documentation

### PR DESCRIPTION
**Pull request by addison-ch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+- [OpenProject Access](#open-project)
+- [SSH Key](#ssh-key-set-up)
+- [Vagrant (Windows Only)](#Vagrant-Environment-Set-Up-for-Windows-Users)
+- [Mac OS](#mac-os-set-up)
+- [Linux](#linux-set-up)
+- [Other Resources](#other-resources)
+
+# Open Project
+
+We use Open Project for task management. (todo: add instructions on how to get access lol)
```
Comment (jlim25) [inline README.md:16]: 1. http://208.68.36.87/
2. Create an account with their school email
3. Drop your school email in the new-members thread under #Firmware on Slack

Then Aryan or Jacky will add them to the OpenProject

File: README.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+# SSH Key Set Up
+
+Please see [this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) for instructions. Feel free to message in #firmware on Slack with any questions!
```
Comment (jlim25) [inline README.md:20]: For complete novices, ssh can be slightly confusing so how about the following?

SSH key is used for access credentials (think of it as your username and password). Without it, you can't contribute to the repository.
1. Create a SSH key by following the instructions [here](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
2. Add the SSH key to your GitHub account by following this [page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)

Note: You can skip this step if you already have a SSH key configured on GitHub

File: README.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+### Getting Started
+
+Make sure everything under **Prerequisites** has been installed, and that you have an SSH key added for your host computer to have GitHub access (check the link above).
+
+Then, run the following commands:
+
+```
+git clone git@github.com:UWaterloo-Formula-Electric/vagrant.git
+cd vagrant
+vagrant up
```
Comment (jlim25) [inline README.md:55]: Can we reduce information overload by removing info that is not super relevant? I have never used any Vagrant commands and I think most people just use VirtualBox to interact with Vagrant. What does everyone else think?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+vagrant reload
+```
+
+(may take a while to complete)
+The commands above clone the vagrant repo which you only need to do this one time in your set up. The next time, you can just call `vagrant up`. When its done, you will see a login window pop up. Do not maximize the window when on the login page (for some reason, this causes it to freeze). Click on the user "vagrant" and login with the password: **vagrant**.
+
+The **shared/** directory in the VM is shared between your host computer (your lapptop) and the virtual environment. The location in your host computer is the path to where you cloned the vagrant repo.
```
Comment (jlim25) [inline README.md:62]: nit: typo for laptop

File: README.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+```
+Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
+Bus 002 Device 002: ID 0483:374b STMicroelectronics ST-LINK/V2.1 (Nucleo-F103RB)
+Bus 002 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
+```
+
+### Clone Firmware
+
+Clone the firmware repository into `shared/` directory. Navigate to where you cloned the **vagrant** repo.
```
Comment (jlim25) [inline README.md:82]: Might want to specify "In vagrant, clone the firmware repo....."

File: README.md (hunk 3 above)
Comment (jlim25) [reply README.md:55]: Generally if we can keep the onboarding short and sweet, newcomers won't find it as confusing and the impression of the firmware team will be better

**Review by jlim25 — CHANGES_REQUESTED** [review]
Some changes are required but overall, pretty good. Thank you Addison for taking this on. We really appreciate it.

File: README.md (hunk 3 above)
Comment (addison-ch) [reply README.md:55]: Interesting, because I've only ever used the vagrant commands 😅

Comment (jlim25) [reply README.md:55]: lol okay maybe keep them in

File: README.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
+# UWFE Firmware
+
+This is the firmware monorepo for the UW Formula Electric team. This repository contains vehicle board firmware, HIL board firmware, and our firmware validation testbed.
```
Comment (Justin-Vuong) [inline README.md:3]: cell tester too

File: README.md (hunk 1 above)
Comment (Justin-Vuong) [reply README.md:16]: step 1: I think the owenbrake.com/pm is a better link lol

File: README.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+6. Clone the firmware repo using the following command: `git clone git@github.com:UWaterloo-Formula-Electric/firmware.git`
+7. Make sure you are in the same directory as the requirements.txt (firmware/common), and run `pip install -r requirements.txt`
+8. Run `make all` within the firmware repo to try and build the code :)
+
+# Other Resources
+
+1. Git tutorial: https://www.freecodecamp.org/news/what-is-git-learn-git-version-control/
+2. CAN Bus tutorial: https://www.kvaser.com/can-protocol-tutorial/
+3. Learning C, plenty of resources online
+4. STM32CubeMX (may be useful) installation: https://www.st.com/en/development-tools/stm32cubemx.html
```
Comment (Justin-Vuong) [inline README.md:175]: Would be really good to have a FAQ at the end here for the things that people bring up this term. Could even ask people to make a PR with their questions :D

File: README.md (hunk 3 above)
Comment (Justin-Vuong) [reply README.md:55]: The virtual box way uses the GUI which is easier IMO but either works

File: README.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+(may take a while to complete)
+The commands above clone the vagrant repo which you only need to do this one time in your set up. The next time, you can just call `vagrant up`. When its done, you will see a login window pop up. Do not maximize the window when on the login page (for some reason, this causes it to freeze). Click on the user "vagrant" and login with the password: **vagrant**.
+
+The **shared/** directory in the VM is shared between your host computer (your lapptop) and the virtual environment. The location in your host computer is the path to where you cloned the vagrant repo.
+
+To turn off the machine, run `vagrant halt` in the same terminal.
+
+### Checking ST Link
```
Comment (Justin-Vuong) [inline README.md:66]: Don't need this, because most people won't have a programmer

File: README.md (hunk 5 above)
Comment (Justin-Vuong) [reply README.md:82]: Agreed, people get very confused on whats in their computer and whats run in the VM

File: README.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+Make sure you are in the same directory as the requirements.txt file (firmware/common) and run the command: `pip3 install -r requirements.txt`
+
+If you are able to successfully run `make all` within the /firmware repository, you have been successful! Feel free to reach out on Slack if you need assistance.
+
+# Linux Set Up
+
+### Installation
+
+1. Install OpenOCD through [this link](https://sourceforge.net/projects/openocd/files/openocd/0.10.0/) or try the following command `sudo apt-get install openocd`
+2. If you want to use OpenOCD from Eclipse, follow the instruction under GNU/Linux (http://gnuarmeclipse.github.io/openocd/install/)
```
Comment (Justin-Vuong) [inline README.md:139]: Remove, don't give this option

File: README.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+3. Download the GNU ARM Embedded Toolchain (version:9-2019-q4-major) from [here](https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads)
+4. Once installed, follow the following commands to ensure you can access the toolchain
+
+```
+# Peep your current PATH
+echo $PATH
+# /usr/local/heroku/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:<BLA BLA BLA>
```
Comment (Justin-Vuong) [inline README.md:146]: Explain how the < BLA BLA BLA > is a placeholder

File: README.md (hunk 5 above)
Comment (aryanGajelli) [reply README.md:82]: I would recommend doing it as "vagrant/shared/"

File: README.md (hunk 3 above)
Comment (aryanGajelli) [reply README.md:55]: I suggest keeping it. I have only used the vagrant commands and would like this option to be available

File: README.md (hunk 1 above)
Comment (jlim25) [reply README.md:16]: wait I never knew that lol

**Review by aryanGajelli — COMMENTED** [review]
Looks good overall. Just need to correct some stuff

**Pull request merged**